src/qml/components/StorageLocations.qml

```diff
@@ -21,25 +21,28 @@ ColumnLayout {
         ButtonGroup.group: group
         text: qsTr("Default")
         description: qsTr("Your application directory.")
+        customDir: optionsModel.getDefaultDataDirString
         recommended: true
         checked: true
     }
     OptionButton {
+        id: customDirOption
         Layout.fillWidth: true
         ButtonGroup.group: group
         text: qsTr("Custom")
         description: qsTr("Choose the directory and storage device.")
+        customDir: fileDialog.folder
         onClicked: fileDialog.open()
```

Comment:
Part of suggestion (2), if the users checks the default option, we don't show the custom data dir anymore.

```suggestion
        onClicked: fileDialog.open()
        onCheckedChanged: {
            if (!customDirOption.checked) {
                customDirOption.customDir = ""
            }
        }
```

Reply:
done